File: bench/type4/adversarial/cases/reduction_minmax_anyall/sum.py
def py_sum_loop(xs):
    total = 0
    for x in xs:
        total += x
    return total


def py_sum_builtin(xs):
    return sum(xs)


def py_product_loop(xs):
    product = 1
    for x in xs:
        product *= x
    return product


def py_wrong_seed(xs):
    total = 1
    for x in xs:
        total += x
    return total

File: bench/type4/adversarial/cases/reduction_minmax_anyall/sum.rs
pub fn rust_sum_loop(xs: &[i64]) -> i64 {
    let mut total = 0;
    for x in xs {
        total += x;
    }
    total
}

pub fn rust_sum_iter(xs: &[i64]) -> i64 {
    xs.iter().sum()
}

pub fn rust_product_iter(xs: &[i64]) -> i64 {
    xs.iter().product()
}
